Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -1333,6 +1333,19 @@ final class DeclarationTests: XCTestCase {
       ]
     )
   }
+
+  func testPrimaryAssociatedTypeNotTerminatedWithAngleBracket() {
+    AssertParse(
+      "protocol1️⃣<2️⃣:3️⃣",
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "expected identifier in protocol"),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "expected name in primary associated type clause"),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "expected '>' to end primary associated type clause"),
+        DiagnosticSpec(locationMarker: "3️⃣", message: "expected type in type"),
```

Comment:
Similar to my PR - `expected type in type` is uh... not great 😅. I'm fine if you just want to add a TODO here

Reply:
FWIW I think I would just not expect this particular diagnostic, the rest seem fine. If we ate the `:` we'd get an "unexpected :" instead which might make more sense.

Reply:
I adjusted a name and now it’s `expected type in inherited type`. I think that makes sense.